File: packages/@juspay-tech/react-native-hyperswitch-klarna/ios/Source/HyperswitchKlarnaViewManager.m
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>

@interface RCT_EXTERN_MODULE(HyperswitchKlarnaPaymentView, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(category, NSString)
RCT_EXPORT_VIEW_PROPERTY(returnURL, NSString)
RCT_EXPORT_VIEW_PROPERTY(onInitialized, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onLoaded, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onAuthorized, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onResize, RCTDirectEventBlock)

RCT_EXTERN_METHOD(initialize:(nonnull NSNumber *)node
                  clientToken:(NSString *)clientToken
                  returnURL:(NSString *)returnURL)

RCT_EXTERN_METHOD(load:(nonnull NSNumber *)node)

RCT_EXTERN_METHOD(authorize:(nonnull NSNumber *)node)

+ (BOOL)requiresMainQueueSetup
{
    return YES;
}

@end

File: packages/@juspay-tech/react-native-hyperswitch-klarna/ios/Source/HyperswitchKlarnaViewManager.swift
import Foundation
import React

@objc(HyperswitchKlarnaPaymentView)
class HyperswitchKlarnaViewManager: RCTViewManager {

    override func view() -> UIView {
        let view = HyperswitchKlarnaView()
        view.bridge = bridge
        return view
    }

    @objc override static func requiresMainQueueSetup() -> Bool {
        return true
    }

    @objc func initialize(_ node: NSNumber, clientToken: String, returnURL: NSString?) {
        DispatchQueue.main.async { [weak self] in
            guard
                let bridge = self?.bridge,
                let view = bridge.uiManager.view(forReactTag: node) as? HyperswitchKlarnaView
            else { return }
            view.initialize(clientToken: clientToken, returnURL: returnURL as String?)
        }
    }

    @objc func load(_ node: NSNumber) {
        DispatchQueue.main.async { [weak self] in
            guard
                let bridge = self?.bridge,
                let view = bridge.uiManager.view(forReactTag: node) as? HyperswitchKlarnaView
            else { return }
            view.load()
        }
    }

    @objc func authorize(_ node: NSNumber) {
        DispatchQueue.main.async { [weak self] in
            guard
                let bridge = self?.bridge,
                let view = bridge.uiManager.view(forReactTag: node) as? HyperswitchKlarnaView
            else { return }
            view.authorize()
        }
    }
}
